src/lib/DeployUtil.ts

```diff
@@ -1144,13 +1144,32 @@ export const deployToJson = (deploy: Deploy) => {
  *
  * @param json
  */
-export const deployFromJson = (json: any) => {
-  const serializer = new TypedJSON(Deploy);
-  const deploy = serializer.parse(json.deploy);
-  if (deploy !== undefined && validateDeploy(deploy).ok) {
-    return deploy;
+export const deployFromJson = (json: any): Result<Deploy, Error> => {
+  if (json.deploy === undefined) {
+    return new Err(new Error("The Deploy JSON doesn't have 'deploy' key."));
+  }
+  let deploy = null;
+  try {
+    const serializer = new TypedJSON(Deploy);
+    deploy = serializer.parse(json.deploy);
+  } catch (serializationError) {
+    return new Err(serializationError);
   }
-  return undefined;
+
+  if (deploy === undefined || deploy === null) {
+    return Err(new Error("Can't parse to JSON"));
```

Comment:
is this the correct error message? it was _passed_ as JSON, it failed to parse as a `Deploy`.

Reply:
Done